Remove AddGeneratorFlag from kubectl create deployment

BLasan opened the pull request:
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This PR replaces the `AddGenratorFlag `function with `AddDryRunFlag` function used in `create_deployment.go` file. And removed the unwanted `AddGenratorFlag` function from utils file

#### Which issue(s) this PR fixes:
Partial fix for #99897

```release-note
Remove the deprecated flag --generator from kubectl create deployment command
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--

-->

/assign @rikatz 
/cc @rikatz

pacoxu commented:
/ok-to-test

rikatz commented:
/retest

I guess this one is good to go:

* AddGeneratorFlags is used only by kubectl create deployment and should actually have been removed with that migration
* The PR #88655 deprecated the flag, and it was released with v1.18 (more than one year, two releases behind).

rikatz commented:
@BLasan on the PR description, at the part of release note, please add something like:

"Remove the deprecated flag --generator from kubectl create deployment command"

BLasan commented:
> @BLasan on the PR description, at the part of release note, please add something like:
> 
> "Remove the deprecated flag --generator from kubectl create deployment command"

Will add sir. Thanks :)

BLasan commented:
> #### What type of PR is this?
> /kind cleanup
> 
> #### What this PR does / why we need it:
> This PR replaces the `AddGenratorFlag `function with `AddDryRunFlag` function used in `create_deployment.go` file. And removed the unwanted `AddGenratorFlag` function from utils file
> 
> #### Which issue(s) this PR fixes:
> Fixes #99897 (Partial)
> 
> #### Special notes for your reviewer:
> #### Does this PR introduce a user-facing change?
> No
> 
> #### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
> /assign @rikatz
> /cc @rikatz



> #### What type of PR is this?
> /kind cleanup
> 
> #### What this PR does / why we need it:
> This PR replaces the `AddGenratorFlag `function with `AddDryRunFlag` function used in `create_deployment.go` file. And removed the unwanted `AddGenratorFlag` function from utils file
> 
> #### Which issue(s) this PR fixes:
> Fixes #99897 (Partial)
> 
> #### Special notes for your reviewer:
> #### Does this PR introduce a user-facing change?
> No
>
>  ### Release note
```
Remove the deprecated flag --generator from kubectl create deployment command
```
> #### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
> /assign @rikatz
> /cc @rikatz

rikatz commented:
you need to edit the PR description :)

BLasan commented:
> you need to edit the PR description :)

Yuh, edited :)

rikatz commented:
hum, bot should have removed the label.

Is the description this way? 

![image](https://user-images.githubusercontent.com/7182341/110245043-71dee680-7f40-11eb-9242-dc095d806be7.png)

It's important to be with the release-note identified to bot recognize this :D

BLasan commented:
> hum, bot should have removed the label.
> 
> Is the description this way?
> 
> ![image](https://user-images.githubusercontent.com/7182341/110245043-71dee680-7f40-11eb-9242-dc095d806be7.png)
> 
> It's important to be with the release-note identified to bot recognize this :D

I removed that part from the PR. Will add this sir. Sorry

rikatz commented:
No need to be sorry :D 

This is the idea of a good first issue, to learn more about how PRs works on Kubernetes :)

BLasan commented:
> No need to be sorry :D
> 
> This is the idea of a good first issue, to learn more about how PRs works on Kubernetes :)

Thanks sir :)

rikatz commented:
/triage accepted
/priority backlog
/lgtm

Will leave the final decision to sig-cli approvers, if this should enter on v1.21 or if we should wait until v1.22 branch is opened :)

Thank you for your contribution!

rikatz commented:
Oh btw some more small nits:

Please change in the description from Fixes #lalal to "Partial fix for #99897", otherwise the original issue will be closed as soon as this gets merged.

And also, still in the "Does this PR introduce a user-facing change?" section there's a leading "No" there :) This needs to be removed, as what matters is what's inside the release-note block that you've changed

BLasan commented:
> #### What type of PR is this?
> /kind cleanup
> 
> #### What this PR does / why we need it:
> This PR replaces the `AddGenratorFlag `function with `AddDryRunFlag` function used in `create_deployment.go` file. And removed the unwanted `AddGenratorFlag` function from utils file
> 
> #### Which issue(s) this PR fixes:
> Fixes #99897 (Partial)
> 
> #### Special notes for your reviewer:
> #### Does this PR introduce a user-facing change?
> No
> 
> ```
> Remove the deprecated flag --generator from kubectl create deployment command
> ```
> 
> #### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
> /assign @rikatz
> /cc @rikatz




> Oh btw some more small nits:
> 
> Please change in the description from Fixes #lalal to "Partial fix for #99897", otherwise the original issue will be closed as soon as this gets merged.
> 
> And also, still in the "Does this PR introduce a user-facing change?" section there's a leading "No" there :) This needs to be removed, as what matters is what's inside the release-note block that you've changed

Looks good sir?

BLasan commented:
> Oh btw some more small nits:
> 
> Please change in the description from Fixes #lalal to "Partial fix for #99897", otherwise the original issue will be closed as soon as this gets merged.
> 
> And also, still in the "Does this PR introduce a user-facing change?" section there's a leading "No" there :) This needs to be removed, as what matters is what's inside the release-note block that you've changed

Looks good sir?

rikatz commented:
yup :) Thanks!

BLasan commented:
> yup :) Thanks!

Thanks for the guidance and support :)

rikatz commented:
/hold

Let's wait until v1.22 cycle starts (we're too close from code freeze right now)

BLasan commented:
> /hold
> 
> Let's wait until v1.22 cycle starts (we're too close from code freeze right now)

Any updates on this PR?

rikatz commented:
/retitle Remove AddGeneratorFlag from kubectl create deployment

BLasan committed:
Remove AddGeneratorFlag function

BLasan commented:
/retest

BLasan commented:
@rikatz Please review

rikatz reviewed (commented):
/lgtm

From my side is good, thank you!

/hold

Will wait the final word from @soltysh if this should enter v1.22 per deprecation policy or not :D

rikatz commented:
/test pull-kubernetes-e2e-kind

soltysh reviewed (approved):
/lgtm
/approve
/hold cancel

soltysh commented:
/retest
